Remove validateDefaultFramebufferAttachment side-effect

lexaknyazev opened the pull request:
#### a523acecfd064fd2d347880d8c72b1ba0e854e72
<pre>
Remove validateDefaultFramebufferAttachment side-effect
<a href="https://bugs.webkit.org/show_bug.cgi?id=242408">https://bugs.webkit.org/show_bug.cgi?id=242408</a>

Reviewed by Kenneth Russell.

This function must not modify its argument
because the caller does not expect that.

Fixes 8 failures in
conformance2/state/gl-object-get-calls.html

* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::validateDefaultFramebufferAttachment):

Canonical link: <a href="https://commits.webkit.org/252898@main">https://commits.webkit.org/252898@main</a>
</pre>

kenrussell reviewed (approved):
Looks good. Is it possible to revise any test expectations after this code change?

lexaknyazev committed:
Remove validateDefaultFramebufferAttachment side-effect
https://bugs.webkit.org/show_bug.cgi?id=242408

Reviewed by Kenneth Russell.

This function must not modify its argument
because the caller does not expect that.

Fixes 8 failures in
conformance2/state/gl-object-get-calls.html

* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::validateDefaultFramebufferAttachment):

Canonical link: https://commits.webkit.org/252898@main
